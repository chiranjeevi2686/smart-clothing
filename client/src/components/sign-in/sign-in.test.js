import React from "react";
import { render, fireEvent } from '@testing-library/react';
import SignIn from './sign-in.component';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const defaultStore = 
    {
        displayName: '',
        name: '',
        email: '',
        password: ''
    };
const mockStore = configureStore();
describe('SignIn component', () => {
    it('should render SignIn', async () => {
        const { asFragment, getByTestId } = render(<Provider store={mockStore(defaultStore)}>
            <SignIn />
        </Provider>)
        expect(asFragment()).toMatchSnapshot();
        const emailInput = getByTestId('email-input')
        const passwordInput = getByTestId('password-input')
        const signInButton = getByTestId('sign-in-button')
        const signInButtonGoogle = getByTestId('sign-in-button-google')
        fireEvent.change(emailInput, {target:{value:'testmail'}})
        fireEvent.change(passwordInput, {target:{value:'testpwd'}})
        fireEvent.click(signInButton)
        fireEvent.click(signInButtonGoogle)
        
    })
})
