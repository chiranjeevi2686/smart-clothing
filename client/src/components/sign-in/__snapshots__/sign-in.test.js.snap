// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component should render SignIn 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ bfYTqA"
  >
    <h2
      class="sc-gsTCUz gnBPzW"
    >
      I already have an account
    </h2>
    <span>
      Sign in with your email and password
    </span>
    <form>
      <div
        class="sc-hKgILt cxolEt"
      >
        <input
          class="sc-eCssSg gHGmMu"
          data-test-id="email-input"
          name="email"
          required=""
          type="email"
          value=""
        />
        <label
          class="sc-jSgupP hRBOdq"
        >
          email
        </label>
      </div>
      <div
        class="sc-hKgILt cxolEt"
      >
        <input
          class="sc-eCssSg gHGmMu"
          data-test-id="password-input"
          name="password"
          required=""
          type="password"
          value=""
        />
        <label
          class="sc-jSgupP hRBOdq"
        >
          password
        </label>
      </div>
      <div
        class="sc-dlfnbm UDbKF"
      >
        <button
          class="sc-gKsewC epXRlq"
          data-test-id="sign-in-button"
          type="submit"
        >
          Sign in
        </button>
        <button
          class="sc-gKsewC dgAxfj"
          data-test-id="sign-in-button-google"
          type="button"
        >
          Sign in with google
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
